src/App.tsx

```diff
@@ -1,39 +1,57 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock/Clock';
+
+interface State {
+  hasClock: boolean;
+  clockName: string;
+}
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  nameId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount(): void {
+    document.addEventListener('contextmenu', this.removeClock);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click', this.addClock);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    this.nameId = window.setInterval(() => {
+      this.setState({
+        clockName: getRandomName(),
+      });
+    }, 3300);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount(): void {
```

Comment:
It's extra, bc TypeScript already knows about React types

```suggestion
  componentWillUnmount() {
```